### ISSUE TITLE
Add spell checking validation to Docs CI

### ISSUE THREAD

**Issue by jonathan-innis** [aws/karpenter-provider-aws#3488]
### Tell us about your request

There are currently a few typos in our docs that are caught by users who are reading through the docs more granularly. Ideally, we should have automation that should catch these typos and prevent the PRs from merging.

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

N/A

### Are you currently working around this issue?

N/A

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by njtran** [comment]
We used to have a spell checker that we ended up removing later https://github.com/aws/karpenter/pull/1057. Maybe instead of doing a github action, we can try and add it in `make presubmit`?

**Comment by jonathan-innis** [comment]
Yeah, I think that would be quite a bit more flexible, since devs could run it locally. There is supposedly a spell-checker that runs against our code, but I also haven't seen that catch everything and/or be very effective

**Comment by jonathan-innis** [comment]
Comments like the ones present in #3757 are good examples of why it would be good to automate this behavior so that a reviewer's time isn't spent catching spelling/grammar errors that are easily made when making changes to docs.
